iolite/client.py

```diff
@@ -16,11 +17,11 @@ class Discovered:
     """ Contains the discovered devices. """
 
     discovered: dict
-    unmapped_devices: dict
+    unmapped_devices: defaultdict
 
     def __init__(self):
         self.discovered = {}
-        self.unmapped_devices = {}
+        self.unmapped_devices = defaultdict(list)
```

Comment:
The defaultdict spares us the `if key in self.unmapped_devices:` test, before appending to the list.
So instead of doing

```python
if device.place_identifier not in self.unmapped_devices:
    self.unmapped_devices[device.place_identifier] = []

self.unmapped_devices[device.place_identifier].append(device)
```

you can just call `self.unmapped_devices[device.place_identifier].append(device)`. If the key is not yet present, the defaultdict adds the key and creates an empty list as a value.

(I changed this when refactoring the `add_device` method.)

Reply:
Ahh nice - Didn't know about this datatype :)